samples/Aridka/Aridka.Server/Controllers/AuthorizationController.cs

```diff
@@ -51,10 +58,40 @@ public async Task<IActionResult> Exchange()
                 identity.AddClaim(Claims.Name, await _applicationManager.GetDisplayNameAsync(application),
                     Destinations.AccessToken, Destinations.IdentityToken);
 
-                return SignIn(new ClaimsPrincipal(identity), OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+                // Set the list of scopes granted to the client application in access_token.
+                var principal = new ClaimsPrincipal(identity);
+                principal.SetScopes(request.GetScopes());
+                principal.SetResources(await _scopeManager.ListResourcesAsync(principal.GetScopes()).ToListAsync());
+
+                foreach (var claim in principal.Claims)
+                {
+                    claim.SetDestinations(GetDestinations(claim));
+                }
+
+                return SignIn(principal, OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
             }
 
             throw new NotImplementedException("The specified grant type is not implemented.");
         }
+
+        private IEnumerable<string> GetDestinations(Claim claim)
+        {
+            // Note: by default, claims are NOT automatically included in the access and identity tokens.
+            // To allow OpenIddict to serialize them, you must attach them a destination, that specifies
+            // whether they should be included in access tokens, in identity tokens or in both.
+
+            return claim.Type switch
+            {
+                Claims.Name or 
+                Claims.Subject 
+                    => ImmutableArray.Create(Destinations.AccessToken, Destinations.IdentityToken),
+
+                // Never include the security stamp in the access and identity tokens, as it's a secret value.
+                "AspNet.Identity.SecurityStamp" 
```

Comment:
Since Identity is not used in this sample, I'd remove that.

Reply:
You mean `"AspNet.Identity.SecurityStamp"` case, only
What about `Name` and `Subject` claims?

---
Not sure, as I understand `Client Credentials Grant` should issue `access_token` only, Is there any case when it should issue `id_token`?

Reply:
Yeah, just `"AspNet.Identity.SecurityStamp"`, keep the other ones.

> * Not sure, as I understand `Client Credentials Grant` should issue `access_token` only, Is there any case when it should issue `id_token`?

You're right that there's no `id_token` in the original OAuth 2.0 specification. And since it's not an OIDC flow, there's officially no id_token in that flow.
That said, as a non-standardized extension, OpenIddict allows returning an id_token for the client credentials flow when the `openid` scope is granted. You're free to ignore it if you don't want to return an identity token.

Reply:
Thanks, I think your comment should be documented with this sample

> There's no id_token in the original OAuth 2.0 specification. And since it's not an OIDC flow, there's officially no id_token in that flow.
That said, as a non-standardized extension, OpenIddict allows returning an id_token for the client credentials flow when the openid scope is granted. You're free to ignore it if you don't want to return an identity token.

Reply:
It's certainly something we can do. Do you want to add it as an XML comment in the sample?

Reply:
Sorry, I am not used to using XML Comments

Reply:
Something like this:

```
// In the original OAuth 2.0 specification, the client credentials grant
// doesn't return an identity token, which is an OpenID Connect concept.
//
// As a non-standardized extension, OpenIddict allows returning an id_token
// to convey information about the client application when the "openid" scope
// is granted (i.e specified when calling principal.SetScopes()). When the "openid"
// scope is not explicitly set, no identity token is returned to the client application.
```

Reply:
Is it ok to be added like this
```cs
// Note: In the original OAuth 2.0 specification, the client credentials grant
// doesn't return an identity token, which is an OpenID Connect concept.
//
// As a non-standardized extension, OpenIddict allows returning an id_token
// to convey information about the client application when the "openid" scope
// is granted (i.e specified when calling principal.SetScopes()). When the "openid"
// scope is not explicitly set, no identity token is returned to the client application.

// Set the list of scopes granted to the client application in access_token.
var principal = new ClaimsPrincipal(identity);
principal.SetScopes(request.GetScopes());
principal.SetResources(await _scopeManager.ListResourcesAsync(principal.GetScopes()).ToListAsync());

```